### PR TITLE
Feature/duckling port

### PR DESCRIPTION
**Pull request by vijay120**
This PR solves #9 by making the system entity recognition service port configurable

### CODE REVIEW

File: mindmeld/components/_config.py [hunk 1]
```diff
@@ -521,7 +522,7 @@
 
 DEFAULT_NLP_CONFIG = {
     'resolve_entities_using_nbest_transcripts': [],
-    'system_entity_recognizer': 'duckling'
+    'system_entity_recognizer': {'type': 'duckling', 'url': 'http://localhost:7151/parse'}
```
Comment (minhtuev) [inline mindmeld/components/_config.py]: Should we pass in DEFAULT_DUCKLING_URL?

File: mindmeld/components/_config.py [hunk 2]
```diff
@@ -547,7 +548,7 @@ def get_app_namespace(app_path):
     return _app_namespace
 
 
-def get_system_entity_recognizer_config(app_path):
+def is_duckling_service(app_path):
     """Returns True if the app config specifies that the system entity recognition
```
Comment (minhtuev) [inline mindmeld/components/_config.py]: Update documentation

File: mindmeld/components/_config.py [hunk 3]
```diff
@@ -547,7 +548,7 @@ def get_app_namespace(app_path):
     return _app_namespace
 
 
-def get_system_entity_recognizer_config(app_path):
+def is_duckling_service(app_path):
```
Comment (minhtuev) [inline mindmeld/components/_config.py]: `is_duckling_configured`?

File: mindmeld/system_entity_recognizer.py [hunk 4]
```diff
@@ -50,7 +47,9 @@ def __init__(self, app_path=None):
                 # The service is turned on by default
                 self.is_service_alive = True
```
Comment (minhtuev) [inline mindmeld/system_entity_recognizer.py]: `is_service_alive` sounds like a misnomer?

File: mindmeld/components/_config.py (hunk 1 above)
Comment (vijay120) [reply mindmeld/components/_config.py]: good point, done

File: tests/test_ser.py [hunk 5]
```diff
@@ -270,3 +271,25 @@ def test_time(query, predicted_texts, predicted_values):
         assert p in response_texts
     for p in predicted_values:
         assert p in response_values
+
+
+def test_system_entity_recognizer_component_no_config(kwik_e_mart_app_path):
+    # If the app has no config, then we need to default to mallard
```
Comment (serapio) [inline tests/test_ser.py]: mallard? what is mallard? 🤔

**Pull request merged**